geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert2/transforms/Expression.scala

```diff
@@ -170,14 +170,17 @@ object Expression {
   }
 
   case class FieldLookup(n: String) extends Expression {
-    private var doEval: EvaluationContext => Any = ctx => {
-      val idx = ctx.indexOf(n)
-      // rewrite the eval to lookup by index
-      doEval = if (idx < 0) { _  => null } else { ec => ec.get(idx) }
-      doEval(ctx)
-    }
 
-    override def eval(args: Array[Any])(implicit ctx: EvaluationContext): Any = doEval(ctx)
+    private var ref: EvaluationContext = _
+    private var i = -1
+
+    override def eval(args: Array[Any])(implicit ctx: EvaluationContext): Any = {
+      if (!ctx.eq(ref)) {
+        ref = ctx
+        i = ctx.indexOf(n)
+      }
+      if (i == -1) { null } else { ctx.get(i) }
```

Comment:
Is caching the evaluationContext actually saving time here?  (I'm kinda fine with it either way.)

Reply:
not sure if it's saving time, but we need to keep a reference to it so we can know if the index of a named value has changed (which it can with a new evaluation context)